packages/admin-ui/src/combobox/combobox-multiple.state.ts

```diff
@@ -18,37 +38,74 @@ export function useComboboxMultipleState(
   })
 
   const checkbox = useCheckboxState({
-    defaultValue: defaultSelected,
+    defaultValue: simplifiedDefaultSelected,
   })
 
   const clearSelected = useCallback(() => {
     combobox.setValue('')
+    setSelectedItems([])
     checkbox.setValue([])
   }, [checkbox.setValue, combobox.setValue])
 
-  const removeSelected = useCallback(
+  const unselect = useCallback(
     (value: string) => {
-      checkbox.setValue((values: string[]) =>
-        values.filter((currentValue) => currentValue !== value)
+      const newVal = selectedItems.filter(
+        (currentValue) => getOptionValue(currentValue) !== value
       )
+
+      setSelectedItems(newVal)
+      checkbox.setValue(newVal.map((i) => getOptionValue(i)))
     },
     [checkbox]
   )
 
+  const addSelectedItem = (newItem: T) => {
+    setSelectedItems((oldVal) => [...oldVal, newItem])
+  }
+
+  const removeSelectedItem = (removedItem: T) => {
+    setSelectedItems((oldValues) => {
+      return oldValues.filter(
+        (currentValue) =>
+          getOptionValue(currentValue) !== getOptionValue(removedItem)
+      )
+    })
+  }
+
   useEffect(() => {
     if (shouldClearOnSelect) combobox.setValue('')
   }, [checkbox.value, shouldClearOnSelect, combobox.setValue])
 
+  useEffect(() => {
+    if (shouldClearOnSelect) combobox.setValue('')
```

Comment:
Same as above? (line 76)

Reply:
Don't know how I ended up with this here 😂